dbms/src/Interpreters/TranslateQualifiedNamesVisitor.cpp

```diff
@@ -74,7 +74,7 @@ void TranslateQualifiedNamesMatcher::visit(ASTIdentifier & identifier, ASTPtr &,
             IdentifierSemantic::setMembership(identifier, best_table_pos + 1);
 
         /// In case if column from the joined table are in source columns, change it's name to qualified.
-        if (best_table_pos && data.source_columns.count(identifier.shortName()))
+        if (best_table_pos && !data.source_columns.empty() && data.source_columns.count(identifier.shortName()))
```

Comment:
How does it change anything? Does the crash happen inside `identifier.shortName()`?

Reply:
It looks like some HACK. The base line means: if identifier belongs to second table, we need a long name for it. And it looks correct without changes.
The issue may be in push down where we need to clear semantics for Identifier before pushing it down.